### ISSUE TITLE
'ClassBalancedDataset' object has no attribute 'evaluate'

### ISSUE THREAD

**Issue by aina1997** [open-mmlab/mmclassification#188]
Hi! I'm trying to use class balanced Dataset, this is my configuration:

`model = dict(
    type='ImageClassifier',
    backbone=dict(
        type='ResNet',
        depth=50,
        num_stages=4,
        out_indices=(3, ),
        style='pytorch'),
    neck=dict(type='GlobalAveragePooling'),
    head=dict(
        type='LinearClsHead',
        num_classes=2,
        in_channels=2048,
        loss=dict(type='CrossEntropyLoss', loss_weight=1.0),
        topk=1))
dataset_type = 'MyDataset'
img_norm_cfg = dict(
    mean=[24.644684636598438, 24.75989356992359, 24.685803467470052],
    std=[10.01540324427892, 10.166633983844736, 10.12125376130067],
    to_rgb=False)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='Normalize',
        mean=[24.644684636598438, 24.75989356992359, 24.685803467470052],
        std=[10.01540324427892, 10.166633983844736, 10.12125376130067],
        to_rgb=False),
    dict(type='Resize', size=224, backend='pillow'),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='Normalize',
        mean=[24.644684636598438, 24.75989356992359, 24.685803467470052],
        std=[10.01540324427892, 10.166633983844736, 10.12125376130067],
        to_rgb=False),
    dict(type='Resize', size=224, backend='pillow'),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='Collect', keys=['img'])
]
data = dict(
    samples_per_gpu=20,
    workers_per_gpu=2,
    train=dict(
        type='ClassBalancedDataset',
        oversample_thr=0.001,
        dataset=dict(
            type='MyDataset',
            pipeline=[
                dict(type='LoadImageFromFile'),
                dict(type='Resize', size=224, backend='pillow'),
                dict(
                    type='Normalize',
                    mean=[
                        24.644684636598438, 24.75989356992359,
                        24.685803467470052
                    ],
                    std=[
                        10.01540324427892, 10.166633983844736,
                        10.12125376130067
                    ],
                    to_rgb=False),
                dict(type='ImageToTensor', keys=['img']),
                dict(type='ToTensor', keys=['gt_label']),
                dict(type='Collect', keys=['img', 'gt_label'])
            ],
            data_prefix=
            '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/classification/training',
            ann_file=
            '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/classification/train.txt'
        )),
    val=dict(
        type='ClassBalancedDataset',
        oversample_thr=0.001,
        dataset=dict(
            type='MyDataset',
            pipeline=[
                dict(type='LoadImageFromFile'),
                dict(type='Resize', size=224, backend='pillow'),
                dict(
                    type='Normalize',
                    mean=[
                        24.644684636598438, 24.75989356992359,
                        24.685803467470052
                    ],
                    std=[
                        10.01540324427892, 10.166633983844736,
                        10.12125376130067
                    ],
                    to_rgb=False),
                dict(type='ImageToTensor', keys=['img']),
                dict(type='Collect', keys=['img'])
            ],
            data_prefix=
            '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/classification/training',
            ann_file=
            '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/classification/val.txt'
        )),
    test=dict(
        type='ClassBalancedDataset',
        oversample_thr=0.001,
        dataset=dict(
            type='MyDataset',
            pipeline=[
                dict(type='LoadImageFromFile'),
                dict(type='Resize', size=224, backend='pillow'),
                dict(
                    type='Normalize',
                    mean=[
                        24.644684636598438, 24.75989356992359,
                        24.685803467470052
                    ],
                    std=[
                        10.01540324427892, 10.166633983844736,
                        10.12125376130067
                    ],
                    to_rgb=False),
                dict(type='ImageToTensor', keys=['img']),
                dict(type='Collect', keys=['img'])
            ],
            data_prefix=
            '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/classification/training',
            ann_file=
            '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/test.txt'
        )))
evaluation = dict(
    interval=1,
    metric=['accuracy', 'precision', 'recall', 'f1_score', 'support'])
checkpoint_config = dict(interval=1)
log_config = dict(interval=200, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=None)
lr_config = dict(policy='step', step=[15])
total_epochs = 20
dataset_A_train = dict(
    type='MyDataset',
    oversample_thr=0.001,
    dataset=dict(
        type='Dataset_A',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='Normalize',
                mean=[
                    24.644684636598438, 24.75989356992359, 24.685803467470052
                ],
                std=[10.01540324427892, 10.166633983844736, 10.12125376130067],
                to_rgb=False),
            dict(type='Resize', size=224, backend='pillow'),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='ToTensor', keys=['gt_label']),
            dict(type='Collect', keys=['img', 'gt_label'])
        ]),
    ann_file=
    '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/classification/labels.txt',
    data_prefix=
    '/home/usuaris/imatge/apalacios/TFM/EUROFUSION-visible/TFM-Marti/results/classification/training',
    pipeline=[
        dict(type='LoadImageFromFile'),
        dict(
            type='Normalize',
            mean=[24.644684636598438, 24.75989356992359, 24.685803467470052],
            std=[10.01540324427892, 10.166633983844736, 10.12125376130067],
            to_rgb=False),
        dict(type='Resize', size=224, backend='pillow'),
        dict(type='ImageToTensor', keys=['img']),
        dict(type='ToTensor', keys=['gt_label']),
        dict(type='Collect', keys=['img', 'gt_label'])
    ])
work_dir = './work_dirs/myconfig'
gpu_ids = range(0, 1)`

But this error apear: 'ClassBalancedDataset' object has no attribute 'evaluate'

And I don't know how to fixe it. At MyDayaset I have a evaluate detinition...

**Comment by LXXXXR** [comment]
Hi @aina1997, 
For testing and validation, please use `MyDataset` instead of `ClassBalancedDataset`. Hope this helps.

**Comment by aina1997** [comment]
Correct! This was the problem! Fixed! Thanks!!!
